Master RMarkdown Document & Render Code/SDC Flag.R

```diff
@@ -0,0 +1,293 @@
+  source("./General Health/3. General Health Outputs.R")
+  ltc_multimorbidity <- ltc_multimorbidity |> 
+    rename('n' = 'people',
+           'measure' = 'total_ltc') |> 
+    mutate(name = 'ltc_multimorbidity')
+  
+  ltc_types <- ltc_types |>
+    rename('n' = 'value',
+           'measure' = 'key') |> 
+    mutate(name = 'ltc types')
+  
+  top5ltc_loc <- top5ltc_loc |> 
+    rename('n' = 'value',
+           'measure' = 'Prevalence') |> 
+    mutate(name = 'top5ltc',
+           age_group = 'all ages')
+    
+  
+  
+  # Define data frames and their corresponding sheet names
+  df3 <- list("Long_Term_Conditions" = ltc_multimorbidity[,c("measure","n","age_group",'name')],
+             "LTC_Types" = ltc_types[,c("measure","n","age_group",'name')],
+             "Top5_LTC" = top5ltc_loc[,c("measure","n","age_group",'name')])
+        
+  
+  
+  
+  for (name in df_names3) {
+    # Get the current dataframe
+    output3 <- df3[[name]]
+    
+    # Add locality name to the output dataframe
+    output3$locality <- LOCALITY
+    
+    # Append the current dataframe to corresponding element in excel_output
+    excel_output3[[name]] <- rbind(excel_output3[[name]], output3)
+  }
+  
+  
+  
+}
+
+
+
+# Apply the function to each dataframe in the list
+flagged_dfs3 <- lapply(excel_output3, flag_low_numbers)
+
+# Extract rows flagged as 1 from each dataframe and combine them
+flagged_rows3 <- do.call(rbind, lapply(flagged_dfs3, function(df) {
+  df[df$flag == 1, ]
+})) |> 
+  select('locality','name','measure','age_group','n')
+
+###Write to excel 
+
+library(openxlsx)
+
+# Define a function to save multiple dataframes to an Excel workbook
+save_dataframes_to_excel <- function(dataframes, sheet_names, file_name) {
+  # Create a new workbook
+  wb <- createWorkbook()
+  
+  # Loop over each dataframe and corresponding sheet name
+  for (i in seq_along(dataframes)) {
+    # Get the current dataframe
+    output <- dataframes[[i]]
+    
+    # Get the sheet name
+    dataframe_name <- sheet_names[[i]]
+    
+    # Create a new sheet with the dataframe name as the sheet name
+    addWorksheet(wb, sheetName = dataframe_name)
+    
+    # Write the dataframe to the current sheet
+    writeData(wb, sheet = dataframe_name, x = output)
+  }
+  
+  # Save the workbook to a file
+  saveWorkbook(wb, file_name, overwrite = TRUE)
+}
+
+# List of dataframes
+excel_output <- list(flagged_rows1,flagged_rows2,flagged_rows3)
+
+# Corresponding sheet names
+excel_names <- c("SMR01_based", "SMR01_Age_Groups", "LTCs_Age_Groups")
+
+# Call the function to save the dataframes to an Excel file
+save_dataframes_to_excel(excel_output, excel_names, paste0("/conf/LIST_analytics/West Hub/02 - Scaled Up Work/RMarkdown/Locality Profiles/background data/SDC/",HSCP,".xlsx"))
+
+
+# Save the workbook to a file
+openxlsx::saveWorkbook(wb, paste0("/conf/LIST_analytics/West Hub/02 - Scaled Up Work/RMarkdown/Locality Profiles/background data/",HSCP,".xlsx"), overwrite = TRUE)
```

Comment:
This looks like it's already done in the `save_dataframes_to_excel` function, are you accidentally saving twice, or have I missed something?